### PR TITLE
Add string attributes(carBrand, carType) to model.customer.Car class

### PR DESCRIPTION
**Pull request by OhJunMing**
Car class now has 2 string attributes carBrand, carType

This would help with future functionality
As we encapsulate carBrand, carType into Car class
Existing
Customer
-phone,name,email,
-carsOwned<Car, coe exp date>

Proposed 1.
Customer
-phone,name,email,
-carsOwned<Car, coe exp date>
-CarPreference<Car>

### CODE REVIEW

File: src/main/java/seedu/address/model/customer/Car.java [hunk 1]
```diff
@@ -4,54 +4,76 @@
...
-                && carId.equals(((Car) other).carId)); // state check
+                && carBrand.equals(((Car) other).carBrand)
+                && carType.equals(((Car) other).carType)); // state check
     }
 
     @Override
     public int hashCode() {
-        return carId.hashCode();
+        // NEED TO VERIFY carbrand or carType hashcode
+        return carBrand.hashCode();
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/model/customer/Car.java:76]: I would prefer if you have hash code account for both carBrand and carType

File: src/main/java/seedu/address/model/customer/Car.java [hunk 2]
```diff
@@ -4,54 +4,76 @@
...
+        checkArgument(isValidCar(carBrand, carType), MESSAGE_CONSTRAINTS);
+
+        this.carBrand = carBrand;
+        this.carType = carType;
+    }
+
+    /**
+     * Returns true if a given string is a valid car.
+     */
+    public static boolean isValidCar(String test1, String test2) {
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/model/customer/Car.java:41]: Method variables `test1` and `test2` should be descriptive, like `testBrand` and `testType`

File: src/main/java/seedu/address/storage/JsonAdaptedCustomer.java [hunk 3]
```diff
@@ -81,9 +81,9 @@ public JsonAdaptedCustomer(Customer source) {
     }
 
     /**
-     * Converts this Jackson-friendly adapted person object into the model's {@code Customer} object.
+     * Converts this Jackson-friendly adapted customer object into the model's {@code Customer} object.
      *
-     * @throws IllegalValueException if there were any data constraints violated in the adapted person.
+     * @throws IllegalValueException if there were any data constraints violated in the adapted customer.
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/storage/JsonAdaptedCustomer.java:86]: While good to see, is it best practice to have a feature PR and a bug-fix PR mixed together? Worried we might be penalized for bad PR practice

**Review by nighoggDatatype — CHANGES_REQUESTED** [review]
Almost looks good to me, some changes and we should be good

File: src/main/java/seedu/address/model/customer/Car.java (hunk 1 above)
Comment (OhJunMing) [reply src/main/java/seedu/address/model/customer/Car.java]: Hey im not really sure abt "hash code account"

I amend to return this.hashcode;
this refers to the car object

File: src/main/java/seedu/address/model/customer/Car.java (hunk 2 above)
Comment (OhJunMing) [reply src/main/java/seedu/address/model/customer/Car.java]: agreed, changed to Car(String carBrand, String carType)

File: src/main/java/seedu/address/storage/JsonAdaptedCustomer.java (hunk 3 above)
Comment (OhJunMing) [reply src/main/java/seedu/address/storage/JsonAdaptedCustomer.java:86]: Yea, should  e no issue, very very minor edit of the naming of the java doc commetns, not like adding test code

**Review by nighoggDatatype — APPROVED** [review]
LGTM

**Comment by OhJunMing** [comment]
WAIT LET ME REFACTOR
AFTER I MERGED SHERMAN PR, there were some conflicts, unintended behaviours

File: src/main/java/seedu/address/model/customer/Car.java (hunk 1 above)
Comment (OhJunMing) [reply src/main/java/seedu/address/model/customer/Car.java]: LOL, this caused recursion, stack overflow, but I corrected it, its works now

**Review by Shizheng001 — APPROVED** [review]

**Pull request merged**